crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -352,6 +352,11 @@ module sui::staking_pool {
             if (total_sui_withdraw_amount >= principal_withdraw_amount)
                 total_sui_withdraw_amount - principal_withdraw_amount
             else 0;
+        // This may happen when we are withdrawing everything from the pool and
+        // the rewards pool balance may be 1 less than reward_withdraw_amount.
+        if (balance::value(&pool.rewards_pool) < reward_withdraw_amount) {
```

Comment:
> The exact reason why this happened needs to be investigated.

I think I would feel better if we did the investigation rather than adding this code--any case where we might be creating money (even if it's just 1 MIST) has to be avoided because it will make our reconciliation of the total SUI supply impossible.

Reply:
However: if we think running in testnet with this code added + emitting an event that will help us debug the issue, I'm ok with that. But I don't think we should plan to keep this code around.